pkg/bootkube/bootstrap.go

```diff
@@ -0,0 +1,84 @@
+package bootkube
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/kubernetes-incubator/bootkube/pkg/asset"
+)
+
+// CreateBootstrapControlPlane seeds static manifests to the kubelet to launch the bootstrap control
+// plane.
+func CreateBootstrapControlPlane(assetDir string, podManifestPath string) error {
+	UserOutput("Running temporary bootstrap control plane...\n")
+
+	// Make secrets temporarily available to bootstrap cluster.
+	if err := os.RemoveAll(asset.BootstrapSecretsDir); err != nil {
+		return err
+	}
+	if err := os.Mkdir(asset.BootstrapSecretsDir, os.FileMode(0755)); err != nil {
+		return err
+	}
+	secretsDir := filepath.Join(assetDir, asset.AssetPathSecrets)
+	secrets, err := ioutil.ReadDir(secretsDir)
+	if err != nil {
+		return err
+	}
+	for _, secret := range secrets {
+		if err := copyFile(filepath.Join(secretsDir, secret.Name()), filepath.Join(asset.BootstrapSecretsDir, secret.Name()), true); err != nil {
+			return err
+		}
+	}
+
+	// Copy the static manifests to the kubelet's pod manifest path.
+	manifestsDir := filepath.Join(assetDir, asset.AssetPathBootstrapManifests)
```

Comment:
I think there's a possible race here when bootstrapping etcd: if the apiserver tries to come up and etcd isn't up yet it will die. We could mitigate this by setting the restartPolicy on the bootstrap api server (and perhaps other pods too) or by waiting for etcd to be ready before bringing up the other pods (this was effectively the previous behavior).